tests/Core/Content/Product/SearchKeyword/ProductSearchBuilderTest.php

```diff
@@ -6,38 +6,83 @@
 
 use FINDOLOGIC\FinSearch\Core\Content\Product\SearchKeyword\ProductSearchBuilder;
 use FINDOLOGIC\FinSearch\Tests\Traits\DataHelpers\SalesChannelHelper;
+use FINDOLOGIC\FinSearch\Utils\Utils;
 use PHPUnit\Framework\MockObject\MockObject;
 use PHPUnit\Framework\TestCase;
+use Shopware\Core\Content\Product\SearchKeyword\ProductSearchBuilderInterface;
 use Shopware\Core\Content\Product\SearchKeyword\ProductSearchTermInterpreterInterface;
 use Shopware\Core\Defaults;
 use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
 use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
 use Symfony\Component\HttpFoundation\Request;
 
 class ProductSearchBuilderTest extends TestCase
 {
     use IntegrationTestBehaviour;
     use SalesChannelHelper;
 
-    public function testSuggestRouteIsIgnoredByFindologic(): void
+    /** @var SalesChannelContext */
+    private $salesChannelContext;
+
+    /** @var MockObject|ProductSearchBuilderInterface */
+    private $productSearchBuilderMock;
+
+    public function setUp(): void
     {
-        /** @var ProductSearchTermInterpreterInterface|MockObject $interpreterMock */
         $interpreterMock = $this->getMockBuilder(ProductSearchTermInterpreterInterface::class)
             ->disableOriginalConstructor()
             ->getMock();
-
-        $productSearchBuilderMock = $this->getMockBuilder(ProductSearchBuilder::class)
-            ->setConstructorArgs([$interpreterMock])
-            ->onlyMethods(['buildParent'])
+        $decoratedProductSearchBuilderMock = $this->getMockBuilder(ProductSearchBuilderInterface::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->productSearchBuilderMock = $this->getMockBuilder(ProductSearchBuilder::class)
+            ->setConstructorArgs([$interpreterMock, $decoratedProductSearchBuilderMock])
+            ->onlyMethods(['buildParent', 'buildShopware63AndLower', 'buildShopware64AndGreater'])
             ->getMock();
 
+        $this->salesChannelContext = $this->buildSalesChannelContext(Defaults::SALES_CHANNEL, 'http://test.de');
+
+        parent::setUp();
+    }
+
+    public function testSuggestRouteIsIgnoredByFindologic(): void
+    {
         // Ensure Shopware handles the request.
-        $productSearchBuilderMock->expects($this->once())->method('buildParent');
+        $this->productSearchBuilderMock->expects($this->once())->method('buildParent');
+        $this->productSearchBuilderMock->expects($this->never())->method('buildShopware63AndLower');
+        $this->productSearchBuilderMock->expects($this->never())->method('buildShopware64AndGreater');
```

Comment:
Some of these expectations are pretty repetitive. Maybe you can move them into their own method such as `setProductSearchBuilderExpectations`.